components/search/search.service.ts

```diff
@@ -29,13 +31,23 @@ export class SearchService<T> {
         return this._optionsLookup;
     }
 
-    public set optionsLookup(lookupFn:LookupFn<T>) {
-        this._optionsLookup = lookupFn;
+    public set optionsLookup(lookupFn:LookupFn<T, boolean>) {
+        this._optionsLookup = (query) => lookupFn(query);
+        if (lookupFn.length == 2) {
+            this._selectedLookup = (query) => lookupFn(query, true);
```

Comment:
Can we not do this check in `writeValue`? - rather than add an extra property onto the search service that does pretty much the same thing `optionsLookup`